web/packages/teleport/src/services/websession/websession.test.ts

```diff
@@ -0,0 +1,42 @@
+/**
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import history from 'teleport/services/history';
+
+import websession from '.';
+import api from '../api';
+
+test('user should be redirected to login even if session delete call fails', async () => {
+  const mockPromise = {
+    then: jest.fn().mockReturnThis(),
+    catch: jest.fn().mockReturnThis(),
+    finally: jest.fn().mockImplementation(callback => {
+      callback();
+      return mockPromise;
+    }),
+    [Symbol.toStringTag]: 'Promise',
+  };
```

Comment:
hrmmm, this part seems kinda hacky. i see we don't have a catch block in `websession.logout` but we force define it here? i'm guessing this was required b/c of the unhandled catch block.

if that's the case, i think we should add a catch block, console log the error (i think it's going to throw a "unhandled rejection" error without a catch block, so might as well handle it ourselves) and then modify the test to represent a more accurate flow:


```ts
  jest.spyOn(api, 'delete').mockRejectedValue(new Error('som error'));
  const goToLoginSpy = jest.spyOn(history, 'goToLogin').mockImplementation();
  jest.spyOn(websession, 'clear').mockImplementation();

  websession.logout();
  await waitFor(() => expect(goToLoginSpy).toHaveBeenCalled());
```